### DISCUSSION TITLE
Power draw is incorrect

### DISCUSSION THREAD

**Discussion by wally007** [NetApp/harvest#748] [Q&A] [answered]
Hello,

I'm using latest Harvest RHEL release, with AFF220 and Ontap 9.9.1p5 with 24 * 3.6TB SSD Disks.
I'm trying to get the power drawn from the node or the whole cluster (as I'm getting it from HPe server using iLO) but i've got some strange figures.

1, How is it possible to draw over the limit of the PSU ?
2, It's impossible that single cluster with 24 SSDs consumes so much power IMHO.
- now i dont have a power meter in the datacenter to really verify, but none of our servers consume anywhere near as much (not even 50%)
According to Harvest collected metrics - the AFF220 consumes 2.6kw using 2*900Watt PSUs.

```
shelf_psu_power_drawn{datacenter="EQX",cluster="AGORA",shelf="1.0",psu_id="1"} 1351
shelf_psu_power_rating{datacenter="EQX",cluster="AGORA",shelf="1.0",psu_id="1"} 913
...............................
shelf_psu_power_drawn{datacenter="EQX",cluster="AGORA",shelf="1.0",psu_id="2"} 1377
shelf_psu_power_rating{datacenter="EQX",cluster="AGORA",shelf="1.0",psu_id="2"} 913
```

With the new AFF250 and Ontap 9.9.1p5, the power_drawn metrics are completely missing
Only "shelf" metrics present are (same harvest/zapi config as AFF220)

```
shelf_labels{datacenter="EQX",cluster="TROY",shelf="1.0",module_type="nsm8e",serial_number="SHJHU2137000287",model="NS224NSM8E",state="online",op_status="normal",vendor_name="NETAPP  "} 1.0
shelf_disk_count{datacenter="EQX",cluster="TROY",shelf="1.0"} 18
shelf_status{datacenter="EQX",cluster="TROY",shelf="1.0"} 0
shelf_new_status{datacenter="EQX",cluster="TROY",shelf="1.0"} 1
```

Thank you

**Comment by cgrinds** [comment, marked as answer]
hi @wally007 indeed those look high. Perhaps you're hitting this known [issue](https://kb.netapp.com/Advice_and_Troubleshooting/Data_Storage_Systems/FAS_Systems/Incorrect_power_drawn_values_displayed_for_IOM12-based_AFF%2F%2FFAS_systems_and_shelves) - thanks to @ruanruijuan for digging up that link.

Looks like some hardware scales that number and some don't.

**Reply by wally007** [reply]
Cool - thank you. For AFF220 that is an easy fix.

AFF250 however doesnt report any values - any idea ? Maybe i should contact Netapp tech support to have this looked at ?

```
TROY::*> storage shelf show -power

           Shelf Name: 1.0
             Stack ID: 1
             Shelf ID: 0
            Shelf UID: 00:a0:98:00:00:52:36:3b
        Serial Number: SHJHU2137000287
          Module Type: NSM8E
                Model: NS224NSM8E
         Shelf Vendor: NETAPP
           Disk Count: 18
      Connection Type: NVMe
          Shelf State: Online
               Status: Normal

Power Supply Units:
                                                     Crest  Power   Reset   PSU              Operational
 ID Type Part#        Serial#         Power Rating   Factor Drawn   Capable Enabled Firmware Status
--- ---- ------------ --------------- -------------- ------ ------- ------- ------- -------- ---------------------
  - -    -            -               -              -      -       -       -       -        -

Voltage Sensors:
    Voltage Operational
 ID     (V) Status
--- ------- ----------------------
  -       - -

Current Sensors:
    Current Operational
 ID    (mA) Status
--- ------- ----------------------
  -       - -

Errors:
------
-


TROY::*>
```

**Comment by ruanruijuan** [comment]
@wally007 From the ONTAP software platform team, here is the answer to the AFF 250.

The output of “storage shelf show -power” from the github link is correct for the AFF A250 and is unrelated to burt 1372079. 
---
To me, it looks like the platform team will need to dig deeper to see if they can update their API to reflect the psu info for this new type of AFF250 that has NVME embedded shelf.

**Comment by ruanruijuan** [comment]
@wally007 the platform team further responded that this is by the design. The embedded shelves do not have a PSU and that's why “storage shelf show -power” does not return a value. They did mention a possible way to get the data:

The only thing I see is “system sensors show” at the BMC which will give you a large table of sensors including data like this:

PSU1_VIN         | 209.040    | Volts      | ok    | na        | 90.480    | 93.600    | 261.040   | 263.120   | na
PSU1_VOUT        | 12.012     | Volts      | ok    | na        | 11.336    | 11.440    | 12.948    | 13.156    | na
PSU1_Curr_IIN    | 1.008      | Amps       | ok    | na        | 0.000     | 0.000     | 9.984     | 12.000    | na
PSU1_IOUT        | 16.000     | Amps       | ok    | na        | 0.000     | 0.000     | 130.000   | 132.000   | na
PSU1_PIN         | 205.900    | Watts * hour | ok    | na        | 7.100     | 14.200    | 1611.700  | 1796.300  | na
PSU1_POUT        | 184.600    | Watts * hour | ok    | na        | 7.100     | 14.200    | 1611.700  | 1796.300  | na
PSU2_VIN         | 209.040    | Volts      | ok    | na        | 90.480    | 93.600    | 261.040   | 263.120   | na
PSU2_VOUT        | 12.012     | Volts      | ok    | na        | 11.336    | 11.440    | 12.948    | 13.156    | na
PSU2_Curr_IIN    | 1.152      | Amps       | ok    | na        | 0.000     | 0.000     | 9.984     | 12.000    | na
PSU2_IOUT        | 18.000     | Amps       | ok    | na        | 0.000     | 0.000     | 130.000   | 132.000   | na
PSU2_PIN         | 241.400    | Watts * hour | ok    | na        | 7.100     | 14.200    | 1611.700  | 1796.300  | na
PSU2_POUT        | 220.100    | Watts * hour | ok    | na        | 7.100     | 14.200    | 1611.700  | 1796.300  | na

**Reply by wally007** [reply]
Hi , thanks for the follow up.

I always thought that `storage shelf show -power` shows (main) system power usage. 
Either way, this is not worth spending too much time on but i'm guessing if this is the case  `storage shelf show -power`  will be obsolete soon enough as most of the systems shipped are going to be NVMe.

**Comment by ruanruijuan** [comment]
Opened ONTAP bug 1511476 [3B] - "Node and shelf power metrics support for all the ONTAP latest hardware models, specifically AFF A250s".
